primitives/primproc/primitiveserver.cpp

```diff
@@ -1598,21 +1598,18 @@ struct BPPHandler
   {
     uint32_t uniqueID, sessionID, stepID;
     BPPMap::iterator it;
-
-    try
-    {
-      bs.advance(sizeof(ISMPacketHeader));
-      bs >> sessionID;
-      bs >> stepID;
-      bs >> uniqueID;
-    }
-    catch (...)
+    if (bs.length() < sizeof(ISMPacketHeader) + sizeof(sessionID) + sizeof(stepID) + sizeof(uniqueID))
```

Comment:
This changes the behavior. 
You can sure have this if but at the same time you better get the try block back.

Reply:
"Exceptions are for exceptional cases". This exception happens so often it renders futile any and all attempts to catch actual error in primitiveserver.

Also, you didn't explain the change in behavior. What is different for code that calls this routine?